### PR TITLE
chore: copy redirect + tests to new api

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This is just the old api server's code translated to TS for the new api. I haven't done much with it, but @moT01 is working on some endpoints that use the redirect logic.

### CODE REVIEW

**Review by Sembauke — APPROVED** [review]
passes locally 👍

File: api/src/utils/allowed-origins.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const allowedOrigins = [
+  'https://www.freecodecamp.dev',
+  'https://www.freecodecamp.org',
+  // pretty sure the rest of these can go?
```
[the remark this answers is not in the document]
Comment (moT01) [reply api/src/utils/allowed-origins.ts:4]: I have not considered it. Something like that would probably work, but I prefer to leave it for another PR for now. The main goal here is to duplicate what's on the current API - I try to not rework things too much.

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply api/src/utils/allowed-origins.ts:4]: Yes, but I prefer the simplicity of this approach. There's nothing to think about.

[the remark this answers is not in the document]
Comment (ShaunSHamilton) [reply api/src/utils/allowed-origins.ts:4]: Of course 👍 . What was I thinking! We need to keep the bugs until v1.

**Review by ojeytonwilliams — COMMENTED** [review]
I can't approve the whole thing, but I do approve of Tom's changes!

File: api/src/utils/redirection.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  }: { returnTo?: string; origin?: string; pathPrefix?: string },
+  _homeLocation = HOME_LOCATION
+) {
+  // coerce to strings, just in case something weird and nefarious is happening
+  // TODO: validate, don't coerce
+  returnTo = '' + returnTo;
+  origin = '' + origin;
+  pathPrefix = '' + pathPrefix;
+  // we add the '/' to prevent returns to
+  // www.freecodecamp.org.somewhere.else.com
```
Comment (raisedadead) [inline api/src/utils/redirection.ts:67]: Do we have a test for this? In case someone nukes the comments or normalizes the slash in a later refactoring.

**Review by moT01 — APPROVED** [review]

**Pull request merged**